src/libostree/ostree-repo-checkout.c

```diff
@@ -643,6 +647,32 @@ checkout_one_file_at (OstreeRepo                        *repo,
 
       need_copy = FALSE;
     }
+  else if (is_overlayfs_whiteout)
+    {
+      const char *name = destination_name + (sizeof (OVERLAYFS_WHITEOUT_PREFIX) - 1);
+
+      if (!name[0])
+        return glnx_throw (error, "Invalid empty overlayfs whiteout '%s'", name);
+
+      g_assert (name[0] != '/'); /* Sanity */
+
+      if (mknodat(destination_dfd, name, (source_mode & ~S_IFMT) | S_IFCHR, (dev_t)0) < 0)
```

Comment:
One thing to verify here is whether we also need to do a `fchmodat` - I think the mode we set is modified by `umask`, so in most cases we always unconditionally chmod.

Reply:
good point I will verify this

Reply:
Right, looking at mknodat documentation:
```
The file mode is modified by the process's umask in the usual way: in the absence of a default ACL, the permissions of the created node are (mode & ~umask).
```

So we also need fchmodat